src/libraries/AssociatedLinkedListSetLib.sol

```diff
@@ -0,0 +1,507 @@
+        TempBytesMemory keyBuffer = _allocateTempKeyBuffer(set, associated);
+
+        StoragePointer sentinelSlot = _mapLookup(keyBuffer, SENTINEL_VALUE);
+        bytes32 val = _load(sentinelSlot);
+        return val == bytes32(0) || isSentinel(val); // either the sentinel is unset, or points to itself
+    }
+
+    /// @notice Get the flags on a value in the set.
+    /// @dev The reserved lower 2 bits will not be returned, as those are reserved for the sentinel and has next
+    /// bit.
+    /// @param set The set containing the value.
+    /// @param associated The address the set is associated with.
+    /// @param value The value to get the flags from.
+    /// @return The flags set on the value.
+    function getFlags(AssociatedLinkedListSet storage set, address associated, SetValue value)
+        internal
+        view
+        returns (uint16)
+    {
+        bytes32 unwrappedKey = SetValue.unwrap(value);
+        TempBytesMemory keyBuffer = _allocateTempKeyBuffer(set, associated);
+        return uint16(uint256(_load(_mapLookup(keyBuffer, unwrappedKey))) & 0xFFFC);
+    }
+
+    /// @notice Check if the flags on a value are enabled.
+    /// @dev The reserved lower 2 bits will be ignored, as those are reserved for the sentinel and has next bit.
+    /// @param set The set containing the value.
+    /// @param associated The address the set is associated with.
+    /// @param value The value to check the flags on.
+    /// @param flags The flags to check.
+    /// @return True if all of the flags are enabled, false otherwise.
+    function flagsEnabled(AssociatedLinkedListSet storage set, address associated, SetValue value, uint16 flags)
+        internal
+        view
+        returns (bool)
+    {
+        flags &= 0xFFFC;
+        return getFlags(set, associated, value) & flags == flags;
+    }
+
+    /// @notice Check if the flags on a value are disabled.
+    /// @dev The reserved lower 2 bits will be ignored, as those are reserved for the sentinel and has next bit.
+    /// @param set The set containing the value.
+    /// @param associated The address the set is associated with.
+    /// @param value The value to check the flags on.
+    /// @param flags The flags to check.
+    /// @return True if all of the flags are disabled, false otherwise.
+    function flagsDisabled(AssociatedLinkedListSet storage set, address associated, SetValue value, uint16 flags)
+        internal
+        view
+        returns (bool)
+    {
+        flags &= 0xFFFC;
+        return ~(getFlags(set, associated, value)) & flags == flags;
+    }
+
+    /// @notice Gets all elements in a set.
+    /// @dev This is an O(n) operation, where n is the number of elements in the set.
+    /// @param set The set to get the elements of.
+    /// @return res An array of all elements in the set.
+    function getAll(AssociatedLinkedListSet storage set, address associated)
```

Comment:
We also implemented pagination, which was a nice feature from web2 APIs. But it's probably a overkill for most of accounts.